### PR TITLE
Use TryAdd to add services rather than Add

### PR DESCRIPTION
**Pull request by kichalla**
@HaoK

### CODE REVIEW

**Comment by msotclas** [comment]
Hi __@kichalla__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Kiran Challa). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla2.msopentech.com.
    </p>

TTYL, MSOTBOT;

**Comment by Eilon** [comment]
:shipit: from me, but also @Tratcher at least for FYI information.

**Comment by davidfowl** [comment]
+1 Why was it changed.

**Comment by HaoK** [comment]
Should probably add a unit test as part of this change to ensure proper lifetime of the memory cache now :)

**Comment by kichalla** [comment]
Ahhh my bad...will fix it...

**Comment by kichalla** [comment]
Fixed and added unit tests. /cc: @Tratcher @HaoK @davidfowl @Eilon

File: test/Microsoft.Framework.Caching.Distributed.Tests/CacheServiceExtensionsTests.cs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        public void AddCaching_RegistersDistributedCacheAsTransient()
+        {
+            // Arrange
+            var services = new ServiceCollection();
+
+            // Act
+            services.AddCaching();
+
+            // Assert
+            var distributedCache = services.Where(desc => desc.ServiceType == typeof(IDistributedCache))
```
Comment (HaoK) [inline test/Microsoft.Framework.Caching.Distributed.Tests/CacheServiceExtensionsTests.cs]: nit: I think you can move the predicate directly into FirstorDefault eliminating the need for the intermediate Where()

File: test/Microsoft.Framework.Caching.Distributed.Tests/CacheServiceExtensionsTests.cs [hunk 2]
```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
```
Comment (kichalla) [inline test/Microsoft.Framework.Caching.Distributed.Tests/CacheServiceExtensionsTests.cs]: will change the copyright header

File: test/Microsoft.Framework.Caching.Distributed.Tests/CacheServiceExtensionsTests.cs (hunk 1 above)
Comment (kichalla) [inline test/Microsoft.Framework.Caching.Distributed.Tests/CacheServiceExtensionsTests.cs]: sure

**Comment by Tratcher** [comment]
:shipit:

**Comment by kichalla** [comment]
commit 999a5d37480e9b85c7daaeeb9e8666667381c6b3

**Pull request closed without merging**